### DISCUSSION TITLE
RFE: Option to reject handshake without ETM or EMS extension

### DISCUSSION THREAD

**Discussion by huiyuexu** [openssl/openssl#21472] [General Discussion]
When the client hello does not include the ETM (encrypt then mac extension) or EMS (extend master secret), server will accept it and the handshake will be successful.

Because "non-AEAD + non-ETM" or "non-EMS" are not secure, can we consider that the server should reject the TLS handshake request?

It should be a problem discussion, just mislabeled as a bug.

**Comment by tom-cosgrove-arm** [comment]
Should this actually be moved into Dicussions?

**Comment by lan1120** [comment]
It is convenient for the application layer to provide a callback after processing the clientHello message. The algorithm suite, ETM, and EMS values can be obtained.
